File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;
        int currentWeight = 0;
        int endTruckCnt = 0;
        int index = 0;
        Deque<Integer> onBridge = new ArrayDeque<>();
        for (int i = 0; i < bridge_length; i++) {
            onBridge.addLast(0);
        }
        while (endTruckCnt < truck_weights.length) {
            answer++;
            int out = onBridge.pollFirst();
            if (out != 0) {
                currentWeight -= out;
                endTruckCnt++;
            }
            if (index >= truck_weights.length) {
                continue;
            }
            if (currentWeight + truck_weights[index] <= weight) {
                currentWeight += truck_weights[index];
                onBridge.addLast(truck_weights[index]);
                index++;
            } else {
                onBridge.addLast(0);
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    bridge_list=[0 for i in range(bridge_length)]
    bridge_w = 0
    while truck_weights != []:
        bridge_w -= bridge_list.pop()
        bridge_list.insert(0,0)
        if bridge_w+truck_weights[0]<=weight:
            bridge_list[0] = truck_weights.pop(0)
            bridge_w+=bridge_list[0]
        answer+=1
    answer+=bridge_length
    return answer